cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/LeaderReplicaPerLiveBrokerGoal.java

```diff
@@ -0,0 +1,511 @@
+import com.linkedin.kafka.cruisecontrol.analyzer.ActionType;
+import com.linkedin.kafka.cruisecontrol.analyzer.BalancingAction;
+import com.linkedin.kafka.cruisecontrol.analyzer.BalancingConstraint;
+import com.linkedin.kafka.cruisecontrol.analyzer.OptimizationOptions;
+import com.linkedin.kafka.cruisecontrol.exception.OptimizationFailureException;
+import com.linkedin.kafka.cruisecontrol.model.Broker;
+import com.linkedin.kafka.cruisecontrol.model.ClusterModel;
+import com.linkedin.kafka.cruisecontrol.model.Replica;
+import com.linkedin.kafka.cruisecontrol.monitor.ModelCompletenessRequirements;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.linkedin.kafka.cruisecontrol.analyzer.ActionAcceptance.*;
+
+
+/**
+ * HARD GOAL: Generate leadership movement and leader replica movement proposals to ensure that each alive broker has
+ * at least a certain number of leader replica of each topic in a configured set of topics
+ */
+public class LeaderReplicaPerLiveBrokerGoal extends AbstractGoal {
+  private static final Logger LOG = LoggerFactory.getLogger(LeaderReplicaPerLiveBrokerGoal.class);
+
+  private Set<String> _mustHaveLeaderReplicaPerBrokerTopics;
+  private int _minLeaderReplicaCountForTopicsMustHaveLeaderReplicaPerBroker;
+
+  public LeaderReplicaPerLiveBrokerGoal() {
+    _mustHaveLeaderReplicaPerBrokerTopics = null;
+    _minLeaderReplicaCountForTopicsMustHaveLeaderReplicaPerBroker = 1;
+  }
+
+  /**
+   * Package private for unit test.
+   */
+  public LeaderReplicaPerLiveBrokerGoal(BalancingConstraint constraint) {
+    _balancingConstraint = constraint;
+  }
+
+  @Override
+  public ClusterModelStatsComparator clusterModelStatsComparator() {
+    return new GoalUtils.HardGoalStatsComparator();
+  }
+
+  @Override
+  public ModelCompletenessRequirements clusterModelCompletenessRequirements() {
+    return new ModelCompletenessRequirements(GoalUtils.MIN_NUM_VALID_WINDOWS_FOR_SELF_HEALING, 0.0, true);
+  }
+
+  @Override
+  public String name() {
+    return LeaderReplicaPerLiveBrokerGoal.class.getSimpleName();
+  }
+
+  @Override
+  public boolean isHardGoal() {
+    return true;
+  }
+
+  /**
+   * Check whether given action is acceptable by this goal. An action is acceptable by a goal if it satisfies
+   * requirements of the goal. Requirements(hard goal): A configured set of topics must have at least one leader
+   * replica on each broker.
+   *
+   * @param action Action to be checked for acceptance.
+   * @param clusterModel The state of the cluster.
+   * @return {@link ActionAcceptance#ACCEPT} if the action is acceptable by this goal,
+   * {@link ActionAcceptance#REPLICA_REJECT} otherwise.
```

Comment:
I changed the implementation to only return `REPLICA_REJECT` since even moving one replica of this action type is not accepted, another action with this type to move a replica of another topic might be accepted. Therefore, only `REPLICA_REJECT` and no `BROKER_REJECT`. Let me know if my understanding is correct.